workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,233 @@
+# Prerequisites
+# 1 - Make sure the az secrets have been added to the environment
+# 2 - Make sure the host IIS server has been prepped to accept incoming WinRM Requests
+#     See the IIS App Pool action README for more details: https://github.com/im-open/iis-service-action/#readme
+
+name: Deploy On Prem Website
+on:
+  workflow_dispatch:
+    inputs:
+      branch-tag-sha:
+        description: The branch, tag or sha of the terraform that should be deployed to the Dev environment.
+        required: true
+      environment:
+        description: The environment to Deploy To.
+        required: true
+
+env:
+  PAGERDUTY_SERVICE_IDS: '[ "${{secrets.PAGERDUTY_SERVICE_ID}}" ]' # TODO:  Add your PD Service IDs here, the svc id usually has a format of P0XXXXX
+  PAGERDUTY_WINDOW_IN_MIN: 30 # TODO:  Verify the length of your PD Maintenance Window
+  PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ github.event.inputs.environment }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+  DEPLOY_BOARD_NUM: '' # TODO: Add the automated deployment board number or remove if not using an automated deployment project board.
+  DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+  PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+  ARTIFACT_NAME: 'Published Site' # TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+  ARTIFACT_ZIP: 'artifacts.zip'
+  DEPLOY_ZIP: 'published_site.zip'
+  IIS_SERVER_CERT_PATH: '' # TODO:  The path to the cert for the IIS server remote management
+  WEBSITE_NAME: '' # TODO:  The unique name of the site particularly for IIS server display
+  WEBSITE_HOST_HEADER: '' # TODO:  The IIS server host header that the site responds to
+  WEBSITE_CERT_PATH: '' # TODO:  The path to the private key cert for the IIS server site binding
+  WEBSITE_CERT_FRIENDLY_NAME: '' # TODO:  The friendly name encoded in the IIS private key cert
+  DEPLOYMENT_FOLDER: '' # TODO:  The local target directory for the web site files, i.e., 'c:\inetpub\website'
```

Comment:
If any of these vary between environments you'll want to update it to use the set-variable-based-on-environment action like this: 
```yml
 # TODO:  Fill in the website name for environments that are used and remove the environments that are not used.
  - uses: im-open/set-variable-based-on-environment@v1.0.0
        with:
          variable-name: 'WEBSITE_NAME'
          current-environment: ${{ needs.get-inputs-from-event.outputs.environment }}
          dev-value: '' # TODO: Add the dev app name or delete if env does not exist
          qa-value: '' # TODO: Add the qa app name or delete if env does not exist
          stage-value: '' # TODO: Add the stage app name or delete if env does not exist
          prod-value: '' # TODO: Add the prod app name or delete if env does not exist
          demo-value: '' # TODO: Add the demo app name or delete if env does not exist
          uat-value: '' # TODO: Add the uat app name or delete if env does not exist
```